### PR TITLE
EUI-6645 Restricted case access

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-6646

### Change description ###
Restricted Case Access - HMCTS Case worker, JOH or Professional users attempts to access restricted case (no access)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,7 +74,7 @@
     "@angular/platform-browser-dynamic": "^11.2.14",
     "@angular/router": "^11.2.14",
     "@edium/fsm": "^2.1.2",
-    "@hmcts/ccd-case-ui-toolkit": "5.0.50-sscs-sit-release",
+    "@hmcts/ccd-case-ui-toolkit": "5.0.49-restricted-case-access-v2",
```
Comment (andywilkinshmcts) [inline package.json:77]: Same concern about version number going backwards

**Review by andywilkinshmcts — DISMISSED** [review]
Looks good, bar the possible issue with toolkit version numbers

File: api/roleAccess/index.ts [hunk 2]
```diff
@@ -72,6 +72,25 @@ export async function getAccessRolesByCaseId(req: EnhancedRequest, res: Response
...
+  try {
+    const response: AxiosResponse = await http.post(fullPath, requestPayload, { headers });
+    const finalRoles: CaseRole[] = mapResponseToCaseRoles(
+      response.data.roleAssignmentResponse,
+      req.body.assignmentId,
+      req
+    );
+    return res.status(response.status).send(finalRoles);
+  } catch (error) {
+    console.log('ERROR', error);
```
Comment (Daniel-Lam) [inline api/roleAccess/index.ts:89]: @johnbenjamin-hmcts - please remove this `console.log()` statement if no longer required

File: src/cases/components/restricted-case-access/restricted-case-access.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [],
+      declarations: [RestrictedCaseAccessComponent]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
```
Comment (Daniel-Lam) [inline src/cases/components/restricted-case-access/restricted-case-access.component.spec.ts:16]: @johnbenjamin-hmcts - is there a reason why there are two `beforeEach()` blocks? Can they be combined?

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        { provide: AllocateRoleService, useValue: mockAllocateService },
+        { provide: WASupportedJurisdictionsService, useValue: mockWASupportedJurisdictionsService },
+        { provide: CaseworkerDataService, useValue: mockCaseworkerDataService },
+        { provide: ActivatedRoute, useValue: mockActivatedRoute }
+      ]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
```
Comment (Daniel-Lam) [inline src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.spec.ts:42]: @johnbenjamin-hmcts - is there a reason why there are two `beforeEach()` blocks? Can they be combined?

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  public caseId: string;
+  public caseRoles: CaseRole[];
+  public caseWorkers: Caseworker[];
+  public idamIds: string[];
+  public restrictedCases: RestrictedCase[];
+
+  constructor(private readonly route: ActivatedRoute,
+              private readonly allocateService: AllocateRoleService,
+              private readonly caseworkerDataService: CaseworkerDataService,
+              private waSupportedJurisdictionsService: WASupportedJurisdictionsService) {
```
Comment (Daniel-Lam) [inline src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts:27]: @johnbenjamin-hmcts - consider marking this service as `readonly` like the others

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+  public ngOnInit(): void {
+    this.caseId = this.route.snapshot.params.cid;
+    this.allocateService.getCaseAccessRolesByCaseId(this.caseId).pipe(
+      switchMap(caseRoles => {
+        this.caseRoles = caseRoles;
+        return of(this.getUniqueIdamIds());
+      }), take(1),
+      switchMap(() => {
+        return this.waSupportedJurisdictionsService.getWASupportedJurisdictions();
```
Comment (Daniel-Lam) [inline src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts:38]: @johnbenjamin-hmcts - single-line expression can be rewritten without `return` keyword or curly braces (concise body)

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      }), take(1),
+      switchMap(jurisdictions => {
+        return this.caseworkerDataService.getCaseworkersForServices(jurisdictions);
```
Comment (Daniel-Lam) [inline src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts:41]: @johnbenjamin-hmcts - as above

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+      }), take(1),
+      switchMap(caseworkers => {
+        return of(this.getRestrictedCases(caseworkers));
```
Comment (Daniel-Lam) [inline src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts:44]: @johnbenjamin-hmcts - as above

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+      })
+    ).subscribe(restrictedCases => {
+      this.restrictedCases = restrictedCases;
```
Comment (Daniel-Lam) [inline src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts:47]: @johnbenjamin-hmcts - as above

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - a few minor observations for you to look at

File: package.json (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply package.json]: I have updated the version.

File: api/roleAccess/index.ts (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply api/roleAccess/index.ts]: Thank you @Daniel-Lam - I removed the `console.log` statement.

File: src/cases/components/restricted-case-access/restricted-case-access.component.spec.ts (hunk 3 above)
Comment (johnbenjamin-hmcts) [reply src/cases/components/restricted-case-access/restricted-case-access.component.spec.ts]: Thank you @Daniel-Lam, I have done the changes as you suggested.

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.spec.ts (hunk 4 above)
Comment (johnbenjamin-hmcts) [reply src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.spec.ts]: Thank you @Daniel-Lam, I have done the changes as you suggested.

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts (hunk 5 above)
Comment (johnbenjamin-hmcts) [reply src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts]: Thank you for pointing this one out @Daniel-Lam, I have done the changes as you suggested.

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts (hunk 6 above)
Comment (johnbenjamin-hmcts) [reply src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts]: Thank you @Daniel-Lam, I also managed to add in the error check.

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts (hunk 7 above)
Comment (johnbenjamin-hmcts) [reply src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts]: Thank you @Daniel-Lam, I have done the changes as you suggested.

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts (hunk 8 above)
Comment (johnbenjamin-hmcts) [reply src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts]: Thank you @Daniel-Lam, I have done the changes as you suggested.

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts (hunk 9 above)
Comment (johnbenjamin-hmcts) [reply src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.ts]: Thank you @Daniel-Lam, I have done the changes as you suggested.

**Review by Daniel-Lam — APPROVED** [review]

**Review by andywilkinshmcts — APPROVED** [review]

File: src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.html [hunk 10]
```diff
@@ -0,0 +1,16 @@
+<div class="govuk-width-container govuk-!-margin-bottom-4">
+  <br/>
+  <div class="hmcts-banner">
+    <svg class="hmcts-banner__icon" fill="currentColor" role="presentation" focusable="false" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 25 25" height="25" width="25">
+      <path d="M13.7,18.5h-2.4v-2.4h2.4V18.5z M12.5,13.7c-0.7,0-1.2-0.5-1.2-1.2V7.7c0-0.7,0.5-1.2,1.2-1.2s1.2,0.5,1.2,1.2v4.8
+C13.7,13.2,13.2,13.7,12.5,13.7z M12.5,0.5c-6.6,0-12,5.4-12,12s5.4,12,12,12s12-5.4,12-12S19.1,0.5,12.5,0.5z"></path></svg>
+    <div class="hmcts-banner__message">
+      <span class="hmcts-banner__assistive">information</span>
+      <span>This case is restricted. The details of the users with access are provided below.</span>
```
Comment (Daniel-Lam) [inline src/cases/containers/restricted-case-access-container/restricted-case-access-container.component.html:9]: @johnbenjamin-hmcts - @ronaldmansveld has advised we should apply the Welsh translation pipe to all pages

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - a few changes required

**Comment by johnbenjamin-hmcts** [comment]
PR https://github.com/hmcts/rpx-xui-webapp/pull/3192 replaces this PR.

**Pull request closed without merging**